### ISSUE TITLE
Can the front-end site only be the Flywire official website

### ISSUE THREAD

**Issue by liuyx599** [ChrisRaven/FlyWire-Fragments#1]
If I were a locally generated neuroglancer front-end service just like [ng-extend](https://github.com/seung-lab/ng-extend) instead of https://ngl.flywire.ai/  Can I use this script?  How can I modify it？

**Comment by ChrisRaven** [comment]
If your local version has the same internal structure as FlyWire (e.g. the same layers' names, the same global object's name (viewer), etc.), then it should be enough to modify the lines 7 and 8. Remove one of them and change the other so the URL matches your local server (don't forget about the asterisk at the end). You might also remove lines 10-12 to prevent from autoupdates, which would overwrite your changes.
However, if your internal structure is different, you might have to do some more modifications. For example, if your 3D view tab's name is different, you have to modify the line 54 and pass there the correct name. Same in 68.
If there are more differences, you might have to also modify the library called Dock.js (available at https://chrisraven.github.io/FlyWire-Dock/Dock.js), which is used internally to retrieve the layers, for example. If you'd modify that library, you would also have to update the line 18 to provide the new path.

**Comment by liuyx599** [comment]
I can tell that, thank you very much.&nbsp; So can other monkey scripts can do the same modifications, such as changing url, tab name, etc.?




---Original---
From: "Krzysztof ***@***.***&gt;
Date: Sat, Sep 23, 2023 19:56 PM
To: ***@***.***&gt;;
Cc: "Achilles ***@***.******@***.***&gt;;
Subject: Re: [ChrisRaven/FlyWire-Fragments] Can the front-end site only be theFlywire official website (Issue #1)




 
If your local version has the same internal structure as FlyWire (e.g. the same layers' names, the same global object's name (viewer), etc.), then it should be enough to modify the lines 7 and 8. Remove one of them and change the other so the URL matches your local server (don't forget about the asterisk at the end). You might also remove lines 10-12 to prevent from autoupdates, which would overwrite your changes.
 However, if your internal structure is different, you might have to do some more modifications. For example, if your 3D view tab's name is different, you have to modify the line 54 and pass there the correct name. Same in 68.
 If there are more differences, you might have to also modify the library called Dock.js (available at https://chrisraven.github.io/FlyWire-Dock/Dock.js), which is used internally to retrieve the layers, for example. If you'd modify that library, you would also have to update the line 18 to provide the new path.

**Comment by ChrisRaven** [comment]
Yes, that should usually do the trick. However, I don't know very well, what changes were made to the FlyWire's Neuroglancer, so there might be some more issues, I'm not aware of.
